strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/RepositoryManagementServiceImpl.java

```diff
@@ -243,21 +244,10 @@ public void undeleteTrash()
             final Map<String, ? extends Repository> repositories = storage.getRepositories();
             for (Repository repository : repositories.values())
             {
-                final String storageId = storage.getId();
-                final String repositoryId = repository.getId();
-
-                try
-                {
-                    if (repository.isTrashEnabled())
-                    {
-                        RootRepositoryPath repositoryPath = repositoryPathResolver.resolve(repository);
-                        RepositoryFiles.undelete(repositoryPath);
-                    }
-                }
-                catch (Exception e)
+                if (repository.isTrashEnabled())
                 {
-                    throw new RuntimeException("Unable to undelete trash for storage " + storageId + " in repository " +
-                                               repositoryId, e);
+                    RootRepositoryPath repositoryPath = repositoryPathResolver.resolve(repository);
+                    ThrowingConsumer.unchecked(RepositoryFiles::undelete).accept(repositoryPath);
```

Comment:
@adavid9  can you please add `throws IOException` and replace original `catch (Exception)` with following:
```
catch (IOException e)
                {
                    throw new ArtifactStorageException("Unable to undelete trash for storage " + storageId + " in repository " +
                                               repositoryId, e);
                }
```